nmma/em/systematicsprior.py

```diff
@@ -0,0 +1,148 @@
+import yaml
+from pathlib import Path
+
+
+class ValidationError(ValueError):
+    def __init__(self, key, message):
+        super().__init__(f"Validation error for '{key}': {message}")
+
+
+ALLOWED_FILTERS = [
+    "u",
```

Comment:
these are the old filter names, and this functionality should be made flexible and with a check on comparing the data filters with the systematics filters in the analysis.py

Reply:
Yes, this will be resolved later, before the final merge.

Reply:
The validation check (for all the grouped filters, individual filters, invalid filters, null (filter independent systematics) are validated) happens at two places, first when the systematics are loaded to the prior dict, there `systematic.main(file)` is called which validates all the inputs, see  `nmma/em/systematicsprior.py`. 

Second check happens in the likelihood as well, only if withTime is True in systematics.yaml